### PR TITLE
feat(upgrade-tests): use local build + multiple "old" versions

### PR DESCRIPTION
**Pull request by samugi**
#### Run upgrade tests using local build

Before this change, the upgrade tests used container images for the old and the new Kong versions. The Lua files from the checked-out repository branch were then installed in the new version container. This only worked if the binaries in the container were compatible with the Lua code.

This changes the upgrade tests so that for the new version, the local build is used instead of a patched container.

#### Run upgrade tests from multiple "old" versions

Currently only 2.8.0 is used as the "old" version to run migration tests, migrating it all the way up to the "new" (current) version. This means that only features that are shared across all versions from "old" to "new" can be tested, e.g. a plugin that is not available in `2.8.0` cannot be configured and used in migration tests.

This change introduces a list of "old_versions" and repeats the tests for each. Tests can use the `OLD_KONG_VERSION` environment variable to determine whether they should execute for the current version.

### Checklist

- [x] The Pull Request has tests
- [x] (no) A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [x] (no) There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-3612

### CODE REVIEW

File: scripts/upgrade-tests/source-versions [hunk 1]
```diff
@@ -0,0 +1,6 @@
+2.8.0
+3.1.0
+3.2.2
+3.3.0
+3.4.0
+3.5.0
```
Comment (samugi) [inline scripts/upgrade-tests/source-versions:6]: temporary list: maybe we don't need to keep all

**Review by hanshuebner — APPROVED** [review]
Looks great, thank you!

File: scripts/upgrade-tests/source-versions (hunk 1 above)
Comment (samugi) [reply scripts/upgrade-tests/source-versions]: Updated to just LTSs ^

**Review by jschmid1 — APPROVED** [review]
cool stuff 🚀 lgtm

**Pull request merged**